components/bl-calendar-heatmap/src/index.js

```diff
@@ -1,133 +1,173 @@
+            enabled     : legend === 'LegendLite',
+            itemSelector: `#legend-label-${ instanceId }`,
+          },
+        ],
+        ...validateCalendarLabel(calendarLabel),
+      ]);
+  }, [cal, cellHeight, cellRadius, cellWidth, dataType, datePropName, defaultDataValue, highlightDate, instanceId,
+    labelHeight, labelPosition, labelRotation, labelWidth, labelOffsetX, labelOffsetY, maxDate, maxValid, minDate,
+    minValid, range, scaleColorRange, scaleColorScheme, scaleOpacityBaseColor, scaleType, sort, startDate,
+    subColorLabel, subColorLabelLogic, subGutter, subLabel, subLabelLogic, subType, textAlign, type, valuePropName,
+    verticalOrientation, sourceDataUrl, data, groupYLogic, groupY, scaleDomain, dynamicDimension, gutter, labelLogic,
+    label, subSort, legend, calendarLabel, animationDuration, theme]);
+
+  useEffect(() => cal.on('click', (event, timestamp, value) => onCellClick({ event, timestamp, value })), []);
+
+  component.goNext = steps => cal.next(steps);
+  component.goPrev = steps => cal.previous(steps);
+  component.jumpTo = (date, reset) => cal.jumpTo(date, reset);
+
+  if (!display) {
+    cal.destroy();
 
-      const calendar = ref.current.querySelector('svg');
-
-      calendar.style.height = `calc(100% - ${ legendRef.current.clientHeight }px)`;
-      calendar.style.width = '100%';
-    }
-
-    return () => {
-      window.removeEventListener('resize', handleResize, false);
-    };
-  }, [ref.current, legendRef.current]);
-
-  const transformDayElement = useCallback((element, value) => {
-    const newElement = React.cloneElement(element, { title: value.date });
-    const rateOfSaturation = getSaturationByCount(maxCount, minCount, value.count);
-    const { h, l } = hslColor;
-
-    newElement.props.fill = `hsl(${ h }, ${ rateOfSaturation }%, ${ l }%)`;
-
-    return newElement;
-  }, [maxCount, minCount, hslColor]);
-
-  if (!display || !newCalendarData) {
     return null;
   }
 
   return (
-    <div ref={ ref } className={ cn('bl-customComponent-calendarHeatmap', classList) } style={ rootStyle }>
-      <div className="wrapper">
-        <CalendarHeatmap
-          values={ newCalendarData }
-          startDate={ shiftDate(today, newCalendarData.length) }
-          endDate={ today }
-          showMonthLabels={ showMonthLabels }
-          showWeekdayLabels={ showWeekdayLabels }
-          monthLabels={ month }
-          weekdayLabels={ weeks }
-          tooltipDataAttrs={ getTooltipData }
-          transformDayElement={ transformDayElement }
-          onClick={ ({ date, count }) => onCellClick({ date, count }) }
-        />
-        <Legend
-          maxCount={ maxCount }
-          minCount={ minCount }
-          hslColor={ hslColor }
-          legendRef={ legendRef }
-          legend={ legend }
-          margin={ legendMargin }
-        />
-      </div>
-      <ReactTooltip/>
-    </div>
+    <>
+      <div
+        ref={ elRef }
+        id={ `cal-heatmap-${ instanceId }` }
```

Comment:
```suggestion
        id={ `bl-cal-heatmap-${ instanceId }` }
```

That is needed for understanding at first look in dev tools that this property is under our control